osu.Game/Rulesets/Scoring/DrainingHealthProcessor.cs

```diff
@@ -134,7 +134,7 @@ protected override void ApplyResultInternal(JudgementResult result)
         {
             base.ApplyResultInternal(result);
 
-            if (IsSimulating && !result.Type.IsBonus())
+            if (IsSimulating && !result.Type.IsBonus() && result.Type.IsScorable())
```

Comment:
What is this for? What does health have to do with scorability?

If this is targeted at the new `HealthBonus` thing I'd just say to revert this and make `IsBonus()` return true for `HealthBonus` instead.

Reply:
Should be fixed.